pkg/filter/network/proxy/factory.go

```diff
@@ -79,7 +79,15 @@ func CreateProxyFactory(conf map[string]interface{}) (api.NetworkFilterChainFact
 	}
 
 	if len(p.ExtendConfig) != 0 {
-		gfcf.extendConfig = protocol.HandleConfig(api.ProtocolName(p.DownstreamProtocol), p.ExtendConfig)
+		gfcf.extendConfig = make(map[api.ProtocolName]interface{})
+		if len(gfcf.protocols) == 1 && gfcf.protocols[0] != protocol.Auto {
```

Comment:
we should add a comment for this branch, otherwise, this looks weird.
```
it's just for backward compatibility, will be removed in the feature.
```

Reply:
Or, we should print a warning message to suggest people use the new format, when found the old format.
eg:
```
if p.ExtendConfig[proto] == nil {
}
```